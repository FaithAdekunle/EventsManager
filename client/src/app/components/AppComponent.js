import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import axios from 'axios';
import NavTab from './TabComponent';
import Main from './MainComponent';
import Helpers from './../Helpers';
import CenterActions from '../actions/centerActions';

/**
 * App component class
 */
class App extends React.Component {
  /**
   * executes after component mounts
   * @returns { void }
   */
  componentDidMount() {
    axios
      .get(`${Helpers.localHost}/centers`)
      .then((response) => {
        CenterActions.updateCentersState(response.data);
      });
  }

  /**
   * renders component in browser
   * @returns { component } to be rendered on the page
   */
  render() {
    return (
      <BrowserRouter>
        <React.Fragment>
          <NavTab />
          <Main />
        </React.Fragment>
      </BrowserRouter>
    );
  }
}
export default App;
