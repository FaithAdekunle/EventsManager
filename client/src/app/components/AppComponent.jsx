import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import NavBarComponent from './NavBarComponent.jsx';
import MainComponent from './MainComponent.jsx';


const AppComponent = () => (
  <BrowserRouter>
    <React.Fragment>
      <NavBarComponent />
      <MainComponent />
    </React.Fragment>
  </BrowserRouter>
);
export default AppComponent;
