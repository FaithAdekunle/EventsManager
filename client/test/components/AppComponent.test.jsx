import React from 'react';
import { shallow } from 'enzyme';
import AppComponent from
  '../../src/app/components/AppComponent.jsx';

describe('App component', () => {
  const wrapper = shallow(<AppComponent />);
  test('unmount comonent', () => {
    wrapper.unmount();
  });
});
