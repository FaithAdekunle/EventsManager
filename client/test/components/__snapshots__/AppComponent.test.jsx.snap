// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should mount component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <UNDEFINED>
        <withRouter(Connect(NavBarComponent)) />
        <withRouter(Connect(MainComponent)) />
      </UNDEFINED>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <withRouter(Connect(NavBarComponent)) />,
          <withRouter(Connect(MainComponent)) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <UNDEFINED>
          <withRouter(Connect(NavBarComponent)) />
          <withRouter(Connect(MainComponent)) />
        </UNDEFINED>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <withRouter(Connect(NavBarComponent)) />,
            <withRouter(Connect(MainComponent)) />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
